encrypt.go

```diff
@@ -187,17 +175,64 @@ func (es *encryptStream) init(version Version, sender BoxSecretKey, receivers []
 		return err
 	}
 
-	// Use the header hash to compute the MAC keys.
-	es.computeMACKeys(sender, receivers)
+	es.macKeys = computeMACKeys(order, sender, receivers, es.headerHash)
 
 	return nil
 }
 
-func (es *encryptStream) computeMACKeys(sender BoxSecretKey, receivers []BoxPublicKey) {
-	for _, receiver := range receivers {
-		macKey := computeMACKey(sender, receiver, es.headerHash)
-		es.macKeys = append(es.macKeys, macKey)
+func computeReceiverOrder(version Version, receiverCount int) []int {
+	switch version.Major {
+	case 1:
+		order := make([]int, receiverCount)
+		for i := 0; i < receiverCount; i++ {
+			order[i] = i
+		}
+		return order
+	case 2:
+		return randomPerm(receiverCount)
```

Comment:
Oh duh does it even make sense to permute the order for encryption mode? The IDs are already public, right?